File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> hash=new HashMap<>();
        int n=nums.length;
        hash.put(0,1);
        int prefixSum=0;
        int count=0;
        for(int i:nums)
        {
            prefixSum+=i;
            if(hash.containsKey(prefixSum-k)){
                count+=hash.get(prefixSum-k);
            }

            hash.put(prefixSum,hash.getOrDefault(prefixSum,0)+1);


        }

        return count;

    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        freq={0:1}

        prefix=0
        ans=0

        for num in nums:
            prefix+=num

            if(prefix-k in freq):
                ans+=freq[prefix-k]
            freq[prefix]=freq.get(prefix,0)+1

        return ans





        